pkg/services/gitopswriter/writer_add_test.go

```diff
@@ -0,0 +1,305 @@
+  namespace: wego-system
+resources:
+- app.yaml
+- bar-gitops-deploy.yaml
+- bar-gitops-source.yaml
+`)
+
+var dummyUserKustomization = []byte(`apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+metadata:
+  name: bar
+  namespace: wego-system
+resources:
+- ../../../apps/bar
+`)
+
+var _ = Describe("Add", func() {
+	var _ = BeforeEach(func() {
+		app = models.Application{
+			Name:                "bar",
+			Namespace:           wego.DefaultNamespace,
+			HelmSourceURL:       "",
+			GitSourceURL:        createRepoURL("ssh://git@github.com/foo/bar.git"),
+			Branch:              "main",
+			Path:                "./kustomize",
+			AutomationType:      models.AutomationTypeKustomize,
+			SourceType:          models.SourceTypeGit,
+			HelmTargetNamespace: "",
+		}
+
+		gitProviders.GetDefaultBranchReturns("main", nil)
+
+		ctx = context.Background()
+	})
+
+	Context("add app with config in app repo", func() {
+		BeforeEach(func() {
+			app.GitSourceURL = createRepoURL("ssh://git@github.com/foo/bar.git")
+			app.ConfigURL = app.GitSourceURL
+			gitOpsDirWriter = createDirWriter()
+
+			gitClient.OpenStub = func(s string) (*gogit.Repository, error) {
+				r, err := gogit.Init(memory.NewStorage(), memfs.New())
+				Expect(err).ShouldNot(HaveOccurred())
+
+				_, err = r.CreateRemote(&config.RemoteConfig{
+					Name: "origin",
+					URLs: []string{"git@github.com:foo/bar.git"},
+				})
+				Expect(err).ShouldNot(HaveOccurred())
+				return r, nil
+			}
+		})
+
+		Describe("generates application goat", func() {
+			It("clones the repo to a temp dir", func() {
+				err := gitOpsDirWriter.AddApplication(ctx, app, "test-cluster", true)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(gitClient.CloneCallCount()).To(Equal(1))
+				_, repoDir, url, branch := gitClient.CloneArgsForCall(0)
+
+				Expect(repoDir).To(ContainSubstring("user-repo-"))
+				Expect(url).To(Equal("ssh://git@github.com/foo/bar.git"))
+				Expect(branch).To(Equal("main"))
+			})
+
+			It("writes the files to the disk", func() {
+				fluxClient.CreateSourceGitReturns(dummyGitSource, nil)
+				fluxClient.CreateKustomizationReturns([]byte("kustomization"), nil)
+
+				err := gitOpsDirWriter.AddApplication(ctx, app, "test-cluster", true)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(gitClient.WriteCallCount()).To(Equal(5))
+
+				path, content := gitClient.WriteArgsForCall(0)
+				Expect(path).To(Equal(".weave-gitops/apps/bar/app.yaml"))
+				Expect(string(content)).To(ContainSubstring("kind: Application"))
+
+				path, content = gitClient.WriteArgsForCall(1)
+				Expect(path).To(Equal(".weave-gitops/apps/bar/bar-gitops-deploy.yaml"))
+				Expect(content).To(Equal([]byte("kustomization")))
+
+				path, content = gitClient.WriteArgsForCall(2)
+				Expect(path).To(Equal(".weave-gitops/apps/bar/bar-gitops-source.yaml"))
+
+				augmented, err := automation.AddWegoIgnore(dummyGitSource)
+				Expect(err).To(BeNil())
+				Expect(content).To(Equal(augmented))
```

Comment:
```suggestion
				Expect(automation.AddWegoIgnore(dummyGitSource)).To(Equal(content))
```
This pattern can be simplified down to a single line - this says the matcher To() should be applied to the first value returned and all other values should be nil or the empty value.